### DISCUSSION TITLE
ReplyKeyboardMarkup in isolation

### DISCUSSION THREAD

**Discussion by Arbiter-S** [python-telegram-bot/python-telegram-bot#4133] [Q&A]
Checking the documents and examples I found that ReplyKeyboardMarkup was always used as the key argument to another method e.g Update.message.reply_text().

I have a bot which has different responses in a ConversationHandler. I want to see if it's possible to ask the clients to remove or add the keyboard in isolation so the same code won't have to be repeated for every single possible response the bot may give.

**Comment by Poolitzer** [comment]
What do you mean with "in isolation". Do you mean edit a message to change the reply keyboard markup? That is not possible.

**Reply by Arbiter-S** [reply]
No I mean a method where we don't have to send a message to update the keyboard and just update it without relying on sending a message.
I know I can send a empty message to update the keyboard but does not look like a good approach.

**Reply by Poolitzer** [reply]
you can not edit a reply markup keyboard message
